lib/Sema/CSRanking.cpp

```diff
@@ -1044,6 +1047,24 @@ SolutionCompareResult ConstraintSystem::compareSolutions(
       }
     }
 
+    // When the MemberImportVisibility feature is enabled, any declaration that
+    // has been imported beats a declaration that has not been imported.
+    if (isMemberImportVisibilityEnabled) {
```

Comment:
I think CSRanking is a too big of a hammer for this, this means that the solver would have produce a valid solution in this case just to rule it out. Instead of doing this you can add a new kind to `ScoreKind` right after `SK_Unavailable` and increase it in `resolveOverload` (we do similar things for sync/async choices for example), this would means that we can rule out solutions that have these overloads sooner.

Reply:
I started looking at this but I don't think it will accomplish my goals. I could be misunderstanding your suggestion, but recall that I don't want the solver to rule these overloads out completely, I just want it to use them as a last resort. If the solver rules these out and instead registers fixes, then the resulting AST will contain error nodes. The error nodes will prevent later stages of typechecking from working with these expressions, which will mean the test cases in `[members_transitive_multifile_access_level.swift](https://github.com/swiftlang/swift/pull/76336#diff-ae330ba7bb6e550c1ead8495ecd992f1d97ce4a4eae10a8f53afbf08d1d7bc5e)` won't be fixed, which is the goal of this PR.

Reply:
> I just want it to use them as a last resort

This is exactly what is going to happen, if there are no other possibilities the solver would form a valid solution with increased scores. This works similarly to ranking but allows us to compare and filter in a lightweight way.

Reply:
For example this would mean that we'd always prefer a solution with the smallest possible number of non-imported declarations in it.

Reply:
Ok, I see!